pennylane_lightning/src/simulators/lightning_kokkos/StateVectorKokkos.hpp

```diff
@@ -1625,21 +783,71 @@ class StateVectorKokkos final
     }
 
   private:
-    using GateFunc = std::function<void(const std::vector<size_t> &, bool,
-                                        const std::vector<fp_t> &)>;
-    using GateMap = std::unordered_map<std::string, GateFunc>;
-    const GateMap gates_;
-
-    using GeneratorFunc = std::function<fp_t(const std::vector<size_t> &, bool,
-                                             const std::vector<fp_t> &)>;
-    using GeneratorMap = std::unordered_map<std::string, GeneratorFunc>;
-    const GeneratorMap generator_;
+    std::map<std::string, GateOperation> gates_indices_;
```

Comment:
```suggestion
    std::unordered_map<std::string, GateOperation> gates_indices_;
```

`unordered_map` is better here than `map`, since we require no explicit ordering. Best case lookup is `O(1)`, compared to `O(log(n))` for map